internal/k8s/client.go

```diff
@@ -216,6 +216,26 @@ func (c *Client) DeploymentIsReady(ctx context.Context, namespace, deployment st
 	return nil
 }
 
+// DaemonsetIsReady returns nil if the daemonset is ready, or a non-nil error otherwise.
```

Comment:
Nit: from the function name alone I'd assume it to return a `bool` rather than an `error`. Could we maybe rename it to something like `CheckDaemonSetStatus`?